backstage/packages/backend/src/index.ts

```diff
@@ -4,6 +4,96 @@ import { githubOrgEntityProviderTransformsExtensionPoint } from '@backstage/plug
 // Import { legacyPlugin } from '@backstage/backend-common';
 import { myVerifiedUserTransformer } from './transformers';
 
+// Sign in users who are not in the catalog
+import { googleAuthenticator } from '@backstage/plugin-auth-backend-module-google-provider';
+import { githubAuthenticator } from '@backstage/plugin-auth-backend-module-github-provider';
+import {
+    stringifyEntityRef,
+    DEFAULT_NAMESPACE,
+} from '@backstage/catalog-model';
+import {
+    authProvidersExtensionPoint,
+    createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import { NotFoundError } from '@backstage/errors';
+
+const customAuth = createBackendModule({
+    // This ID must be exactly "auth" because that's the plugin it targets
+    pluginId: 'auth',
+    // This ID must be unique, but can be anything
+    moduleId: 'custom-google-auth-provider', // todo: change this to something more descriptive
+    register(reg) {
+        reg.registerInit({
+            deps: {
+                providers: authProvidersExtensionPoint,
+            },
+            async init({ providers }) {
+                providers.registerProvider({
+                    // This ID must match the actual provider config, e.g. addressing
+                    // auth.providers.github means that this must be "github".
+                    providerId: 'google',
+                    // Use createProxyAuthProviderFactory instead if it's one of the proxy
```

Comment:
this comment confuses me more than it explains. We *are* using OAuth, so this is the correct function, right?